Mathlib/Data/Sym/Sym2.lean

```diff
@@ -692,4 +692,31 @@ instance [Nontrivial α] : Nontrivial (Sym2 α) :=
 unsafe instance [Repr α] : Repr (Sym2 α) where
   reprPrec s _ := f!"s({repr s.unquot.1}, {repr s.unquot.2})"
 
+/--
+Partial map. If `f : Π a, p a → β` is a partial function defined on `a : α` satisfying `p`,
+then `pmap f s h` is essentially the same as `map f s` but is defined only when all members of `s`
+satisfy `p`, using the proof to apply `f`.
+-/
+def pmap {P : α → Prop} (f : ∀ a, P a → β) (s : Sym2 α):
```

Comment:
```suggestion
def pmap {P : α → Prop} (f : ∀ a, P a → β) (s : Sym2 α) :
```
Also, can this be put closer to `map` rather than at the end of the file?